### PR TITLE
api: add attention to HeaderValueOption.append to avoid removing this field

### PR DESCRIPTION
**Pull request by wbpcode**
Signed-off-by: wbpcode <wangbaiping@corp.netease.com>

Commit Message: api: add attention to HeaderValueOption.append to avoid removing this field
Additional Description:

See #22845.

Risk Level: n/a.
Testing: n/a.
Docs Changes: added.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

File: api/envoy/config/core/v3/base.proto [hunk 1]
```diff
@@ -358,6 +358,17 @@ message HeaderValueOption {
...
+  // .. attention::
+  //   The :ref:`ext_authz filter <envoy_v3_api_msg_extensions.filters.http.ext_authz.v3.ExtAuthz>` and
+  //   :ref:`ext_proc filter <envoy_v3_api_msg_extensions.filters.http.ext_proc.v3.ExternalProcessor>` have
+  //   error default value (false) for this field.
+  //   And the default value of
+  //   :ref:`new append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>`
+  //   is always APPEND_IF_EXISTS_OR_ADD.
+  //   It means if we remove this field, the default behavior in these filter will be changed and may break our
+  //   users.
+  //   So we should keep this field and couldn't remove it until the above issue is resolved one day.
```
Comment (soulxu) [inline api/envoy/config/core/v3/base.proto:371]: Emm...The last two sentences seems like a note for the developer, it shouldn't be part of the API document.

File: api/envoy/config/core/v3/base.proto [hunk 2]
```diff
@@ -358,6 +358,17 @@ message HeaderValueOption {
   // existing values. Otherwise it replaces any existing values.
   // This field is deprecated and please use
   // :ref:`append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>` as replacement.
+  //
+  // .. attention::
+  //   The :ref:`ext_authz filter <envoy_v3_api_msg_extensions.filters.http.ext_authz.v3.ExtAuthz>` and
+  //   :ref:`ext_proc filter <envoy_v3_api_msg_extensions.filters.http.ext_proc.v3.ExternalProcessor>` have
+  //   error default value (false) for this field.
```
Comment (phlax) [inline api/envoy/config/core/v3/base.proto:365]:
```suggestion
  //   error default value (``false``) for this field.
```

File: api/envoy/config/core/v3/base.proto [hunk 3]
```diff
@@ -358,6 +358,17 @@ message HeaderValueOption {
...
+  //   And the default value of
+  //   :ref:`new append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>`
+  //   is always APPEND_IF_EXISTS_OR_ADD.
```
Comment (phlax) [inline api/envoy/config/core/v3/base.proto:368]:
```suggestion
  //   is always ``APPEND_IF_EXISTS_OR_ADD``.
```

**Review by phlax — COMMENTED** [review]
couple of format nits inline...

File: api/envoy/config/core/v3/base.proto (hunk 1 above)
Comment (mattklein123) [reply api/envoy/config/core/v3/base.proto:371]: As I mentioned in Slack, at this point the APIs are forever. We are never removing anything, so I think we can remove this. Can we rewrite this as end user documentation? Thank you.

/wait

Comment (wbpcode) [reply api/envoy/config/core/v3/base.proto:371]: It's easy to do some comments update. But I think there still is some info gap. 🤣 I re-read the API version policy. There are some conclusions by my understanding. Feel free to point int out if I am wrong.

1. When we say `the APIs are forever`, it means that this field is forever in the `core.v3.HeaderValueOption`.
2. One day the v4 APIs is enabled, then this field will be removed from the `core.v4.HeaderValueOption`. But the break changes are acceptable for major API versions update.

File: api/envoy/config/core/v3/base.proto [hunk 4]
```diff
@@ -358,6 +358,16 @@ message HeaderValueOption {
...
+  //
+  // .. attention::
+  //   The :ref:`external authorization service <envoy_v3_api_msg_service.auth.v3.CheckResponse>` and
+  //   :ref:`external processor service <envoy_v3_api_msg_service.ext_proc.v3.ProcessingResponse>` have
+  //   error default value (``false``) for this field.
+  //   And the default value of field
+  //   :ref:`append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>`
+  //   is always ``APPEND_IF_EXISTS_OR_ADD``.
+  //   It means that the default behavior of ``HeaderValueOption`` in these two services will be changed after
+  //   this field is removed in the future version of API.
```
Comment (mattklein123) [inline api/envoy/config/core/v3/base.proto:370]: Per other comments, it's not going to get removed, so let's remove this. It's still not clear to me what we are telling the end user. Are we just telling them that the default is false when used in these 2 services? If so can we just say that clearly?

File: api/envoy/config/core/v3/base.proto [hunk 5]
```diff
@@ -358,6 +358,16 @@ message HeaderValueOption {
   // existing values. Otherwise it replaces any existing values.
   // This field is deprecated and please use
   // :ref:`append_action <envoy_v3_api_field_config.core.v3.HeaderValueOption.append_action>` as replacement.
+  //
+  // .. attention::
+  //   The :ref:`external authorization service <envoy_v3_api_msg_service.auth.v3.CheckResponse>` and
+  //   :ref:`external processor service <envoy_v3_api_msg_service.ext_proc.v3.ProcessingResponse>` have
+  //   error default value (``false``) for this field.
```
Comment (mattklein123) [inline api/envoy/config/core/v3/base.proto:365]:
```suggestion
  //   default value (``false``) for this field.
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
/wait

File: api/envoy/config/core/v3/base.proto (hunk 1 above)
Comment (mattklein123) [reply api/envoy/config/core/v3/base.proto]: I haven't gotten around to doing an official policy update, but at this point realistically a) we are never doing v3 and b) we are never actually deleting any fields/code from Envoy. It's too disruptive. cc @envoyproxy/api-shepherds

Comment (wbpcode) [reply api/envoy/config/core/v3/base.proto]: got it.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**